### DISCUSSION TITLE
Please add how to use Vanilla-Extract in ory/elements examples

### DISCUSSION THREAD

**Discussion by meotimdihia** [ory/.github#67] [Ideas]
Currently, I don't know how to implement a custom style:

https://github.com/ory/elements#strongly-typed-css-using-vanilla-extract

Please add the code to Next/React examples: 

![image](https://github.com/ory/.github/assets/300961/e4ffd52a-d676-498d-af7e-16d74b0315bb)

Also currently, it is hard to change the style in @ory/elements. It is good if we can add a headless UI.

**Comment by aeneasr** [comment]
@Benehiko can you take a look?

**Comment by Benehiko** [comment]
Hi @meotimdihia,

Thank you for using `@ory/elements` :) 

You can override the css properties by passing in your own class. to the `AuthCard` component. It's not perfect since we haven't abstracted enough of the components from the styling. But it should also be possible for you to build your own "AuthCard" with your own buttons by using the internal components. `FilterFlowNodes` should be your friend here since it only adds gaps to each element it adds to the dom. The container managing these elements can be whatever you want it to be.  https://github.com/ory/elements/blob/main/src/react-components/ory/helpers/filter-flow-nodes.tsx#L16

On our website, we just add a `max-width` override with tailwindcss (https://www.ory.sh/identity-authentication/)

![image](https://github.com/ory/.github/assets/18033717/6ab83c0e-ee3f-476c-b0cc-63bab49683e5)

```tsx
const AuthComponent => (
 <ThemeProvider>
      <UserAuthCard
        className={cn("!max-w-[28rem]", styles)}
        title={theme.cardTitle}
        flow={theme.flowData || (defaultLogin as SelfServiceFlow)}
        flowType="login"
      />
    </ThemeProvider>
  )
)
```

If you have other suggestions on how we can improve `@ory/elements`, please add an issue to the GitHub repository here https://github.com/ory/elements/issues. Contributions in the form of a PR are also welcome :)

**Reply by meotimdihia** [reply]
@Benehiko Thanks, it works for me. Yesterday, I tried with `w-[40rem]` but forgot about `max-width` limiting this attribute.
Then I read about `Vanilla-Extract` and got confused.

**Reply by meotimdihia** [reply]
@Benehiko Please add an example of how to add class/change style for child elements.
![image](https://github.com/ory/.github/assets/300961/fca82379-e50a-4a90-94bd-13160ff7c1bd)

**Reply by Benehiko** [reply]
You can't at such a nested level. At that level you need to then implement your own `UserAuthCard` using the internal components.  https://github.com/ory/elements/tree/main/src/react-components
